core/src/main/scala/org/apache/spark/status/AppStatusStore.scala

```diff
@@ -113,10 +113,15 @@ private[spark] class AppStatusStore(
     }
   }
 
-  def stageData(stageId: Int, details: Boolean = false): Seq[v1.StageData] = {
+  def stageData(
+    stageId: Int,
+    details: Boolean = false,
+    withSummaries: Boolean = false,
```

Comment:
> Do we need both params? doesn't this have summaries iff quantiles isn't empty?

two params is necessary, yea, we need a check if quantiles is empty, we apply a default value

Reply:
I don't follow - when would say withSummaries be true and quantiles be empty for example? or when is it false while quantiles is non-empty?

Reply:
> when would say withSummaries be true and quantiles be empty for example?
```
?withSummaries=true&quantiles=
```
Maybe this case,  I need to try and test, then update the code.

> when is it false while quantiles is non-empty?

We just ignore this, since quantiles only effect when withSummaries=true

Reply:
![image](https://user-images.githubusercontent.com/46485123/111254942-194ece00-8651-11eb-9428-1272bbb913af.png)

We  don't need to worry about `quantiles ` is empty since we will throw exception.

Reply:
OK, I'm still not sure why there are two parameters? sorry if I'm missing the obvious here.

Reply:
> OK, I'm still not sure why there are two parameters? sorry if I'm missing the obvious here.

You mean `details` and `withSummaries`?  Since we add these parameter in restful path `api/v1/applications/[app-id]/stages`

Reply:
Right, but, why add two parameters to begin with? what are the possible values of withSummaries that aren't implied by the value of details?

Reply:
> Right, but, why add two parameters to begin with? what are the possible values of withSummaries that aren't implied by the value of details?

Hmm, one for metrics distribution summaries one for task details. They work in different ways and do not affect each other.

Add tow parameters have two concern.
1. Some use may just want summaries metrics to get distribution and check data skew but don't want task details data.
2. `details` parameter keeps consistence with origin.

Reply:
OK I see it now, withSummaries causes more info to be returned

Reply:
> OK I see it now, withSummaries causes more info to be returned

Yea, return more summaries metrics in distribution. It's very useful.